Fix issues reported by staticcheck

```
$GOPATH/bin/staticcheck ./cxgo/*
-: main module (github.com/skycoin/cx) does not contain package github.com/skycoin/cx/cxgo/notes.md (compile)
-: no Go files in /Users/emin/dev/upwork/skycoin/cx/cxgo/util (compile)
cxgo/actions/functions.go:129:66: should use make(map[string]*cxcore.CXArgument) instead (S1019)
cxgo/actions/functions.go:133:59: should use make(map[string]bool) instead (S1019)
cxgo/actions/functions.go:140:68: should use make(map[string]*cxcore.CXArgument) instead (S1019)
cxgo/actions/functions.go:969:68: unnecessary use of fmt.Sprintf (S1039)
cxgo/actions/functions.go:975:68: unnecessary use of fmt.Sprintf (S1039)
cxgo/actions/functions.go:998:68: unnecessary use of fmt.Sprintf (S1039)
cxgo/actions/functions.go:1004:68: unnecessary use of fmt.Sprintf (S1039)
cxgo/actions/misc.go:416:3: should merge variable declaration with assignment on next line (S1021)
cxgo/cxgo/cxparser.go:102:15: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (S1007)
cxgo/cxgo/cxparser.go:104:17: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (S1007)
cxgo/cxgo/cxparser.go:107:16: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (S1007)
cxgo/cxgo0/scanner.go:855:4: empty branch (SA9003)
cxgo/cxlexer/parse.go:51:11: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (S1007)
cxgo/cxlexer/parse.go:53:11: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (S1007)
cxgo/cxlexer/parse.go:56:10: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (S1007)
cxgo/cxlexer/parse.go:346:23: should use tmp.String() instead of string(tmp.Bytes()) (S1030)
cxgo/parser/lexer.go:7:2: should not use dot imports (ST1001)
cxgo/webapi/httputil.go:17:3: Errorf is a pure function but its return value is ignored (SA4017)
cxgo/webapi/httputil.go:17:3: error strings should not be capitalized (ST1005)
```

--- a/cxgo/actions/functions.go
+++ b/cxgo/actions/functions.go
@@ -126,18 +126,18 @@ func FunctionDeclaration(fn *cxcore.CXFunction, inputs, outputs []*cxcore.CXArgu
 	var symbols *[]map[string]*cxcore.CXArgument
 	tmp := make([]map[string]*cxcore.CXArgument, 0)
 	symbols = &tmp
-	*symbols = append(*symbols, make(map[string]*cxcore.CXArgument, 0))
+	*symbols = append(*symbols, make(map[string]*cxcore.CXArgument))
 
 	// this variable only handles the difference between local and global scopes
 	// local being function constrained variables, and global being global variables
-	var symbolsScope map[string]bool = make(map[string]bool, 0)
+	var symbolsScope map[string]bool = make(map[string]bool)
 
 	FunctionProcessParameters(symbols, &symbolsScope, &offset, fn, fn.Inputs)
 	FunctionProcessParameters(symbols, &symbolsScope, &offset, fn, fn.Outputs)
 
 	for i, expr := range fn.Expressions {
 		if expr.ScopeOperation == cxcore.SCOPE_NEW {
-			*symbols = append(*symbols, make(map[string]*cxcore.CXArgument, 0))
+			*symbols = append(*symbols, make(map[string]*cxcore.CXArgument))
 		}
 
 		ProcessMethodCall(expr, symbols, &offset, true)
@@ -696,7 +696,7 @@ func ProcessSlice(inp *cxcore.CXArgument) {
 	if elt.IsSlice && len(elt.DereferenceOperations) > 0 && elt.DereferenceOperations[len(elt.DereferenceOperations)-1] == cxcore.DEREF_POINTER {
 		elt.DereferenceOperations = elt.DereferenceOperations[:len(elt.DereferenceOperations)-1]
 		return
-	} 
+	}
 
 	if elt.IsSlice && len(elt.DereferenceOperations) > 0 && len(inp.Fields) == 0 {
 		return
@@ -966,13 +966,13 @@ func CopyArgFields(sym *cxcore.CXArgument, arg *cxcore.CXArgument) {
 					if declSpec[len(declSpec)-1] == cxcore.DECL_ARRAY || declSpec[len(declSpec)-1] == cxcore.DECL_SLICE {
 						declSpec = declSpec[:len(declSpec)-1]
 					} else {
-						println(cxcore.CompilationError(sym.FileName, sym.FileLine), fmt.Sprintf("invalid indexing"))
+						println(cxcore.CompilationError(sym.FileName, sym.FileLine), "invalid indexing")
 					}
 				case cxcore.DECL_DEREF:
 					if declSpec[len(declSpec)-1] == cxcore.DECL_POINTER {
 						declSpec = declSpec[:len(declSpec)-1]
 					} else {
-						println(cxcore.CompilationError(sym.FileName, sym.FileLine), fmt.Sprintf("invalid indirection"))
+						println(cxcore.CompilationError(sym.FileName, sym.FileLine), "invalid indirection")
 					}
 				default:
 					declSpec = append(declSpec, elt.DeclarationSpecifiers[c])
@@ -995,13 +995,13 @@ func CopyArgFields(sym *cxcore.CXArgument, arg *cxcore.CXArgument) {
 					if declSpec[len(declSpec)-1] == cxcore.DECL_ARRAY || declSpec[len(declSpec)-1] == cxcore.DECL_SLICE {
 						declSpec = declSpec[:len(declSpec)-1]
 					} else {
-						println(cxcore.CompilationError(sym.FileName, sym.FileLine), fmt.Sprintf("invalid indexing"))
+						println(cxcore.CompilationError(sym.FileName, sym.FileLine), "invalid indexing")
 					}
 				case cxcore.DECL_DEREF:
 					if declSpec[len(declSpec)-1] == cxcore.DECL_POINTER {
 						declSpec = declSpec[:len(declSpec)-1]
 					} else {
-						println(cxcore.CompilationError(sym.FileName, sym.FileLine), fmt.Sprintf("invalid indirection"))
+						println(cxcore.CompilationError(sym.FileName, sym.FileLine), "invalid indirection")
 					}
 				case cxcore.DECL_POINTER:
 					if sym.FileLine != arg.FileLine {

--- a/cxgo/actions/misc.go
+++ b/cxgo/actions/misc.go
@@ -413,9 +413,7 @@ func WritePrimary(typ int, byts []byte, isGlobal bool) []*cxcore.CXExpression {
 		arg.AddType(cxcore.TypeNames[typ])
 		arg.Package = pkg
 
-		var size int
-
-		size = len(byts)
+		var size = len(byts)
 
 		arg.Size = cxcore.GetArgSize(typ)
 		arg.TotalSize = size

--- a/cxgo/cxgo/cxparser.go
+++ b/cxgo/cxgo/cxparser.go
@@ -99,18 +99,18 @@ func lexerStep0(srcStrs, srcNames []string) int {
 	reComment := regexp.MustCompile("//")
 
 	rePkg := regexp.MustCompile("package")
-	rePkgName := regexp.MustCompile("(^|[\\s])package\\s+([_a-zA-Z][_a-zA-Z0-9]*)")
+	rePkgName := regexp.MustCompile(`(^|[\s])package\s+([_a-zA-Z][_a-zA-Z0-9]*)`)
 	reStrct := regexp.MustCompile("type")
-	reStrctName := regexp.MustCompile("(^|[\\s])type\\s+([_a-zA-Z][_a-zA-Z0-9]*)?\\s")
+	reStrctName := regexp.MustCompile(`(^|[\s])type\s+([_a-zA-Z][_a-zA-Z0-9]*)?\s`)
 
 	reGlbl := regexp.MustCompile("var")
-	reGlblName := regexp.MustCompile("(^|[\\s])var\\s([_a-zA-Z][_a-zA-Z0-9]*)")
+	reGlblName := regexp.MustCompile(`(^|[\s])var\s([_a-zA-Z][_a-zA-Z0-9]*)`)
 
 	reBodyOpen := regexp.MustCompile("{")
 	reBodyClose := regexp.MustCompile("}")
 
 	reImp := regexp.MustCompile("import")
-	reImpName := regexp.MustCompile("(^|[\\s])import\\s+\"([_a-zA-Z][_a-zA-Z0-9/-]*)\"")
+	reImpName := regexp.MustCompile(`(^|[\s])import\\s+"([_a-zA-Z][_a-zA-Z0-9/-]*)"`)
 
 	profiling.StartProfile("1. packages/structs")
 	// 1. Identify all the packages and structs

--- a/cxgo/cxlexer/parse.go
+++ b/cxgo/cxlexer/parse.go
@@ -14,8 +14,8 @@ import (
 	"github.com/skycoin/cx/cx"
 	"github.com/skycoin/cx/cxgo/actions"
 	"github.com/skycoin/cx/cxgo/cxgo0"
-	"github.com/skycoin/cx/cxgo/util/cxprof"
 	"github.com/skycoin/cx/cxgo/parser"
+	"github.com/skycoin/cx/cxgo/util/cxprof"
 )
 
 // re contains all regular expressions used for lexing
@@ -48,18 +48,18 @@ var re = struct {
 	multiCommentClose: regexp.MustCompile(`\*/`),
 
 	pkg:     regexp.MustCompile("package"),
-	pkgName: regexp.MustCompile("(^|[\\s])package\\s+([_a-zA-Z][_a-zA-Z0-9]*)"),
+	pkgName: regexp.MustCompile(`(^|[\s])package\s+([_a-zA-Z][_a-zA-Z0-9]*)`),
 	str:     regexp.MustCompile("type"),
-	strName: regexp.MustCompile("(^|[\\s])type\\s+([_a-zA-Z][_a-zA-Z0-9]*)?\\s"),
+	strName: regexp.MustCompile(`(^|[\s])type\s+([_a-zA-Z][_a-zA-Z0-9]*)?\s`),
 
 	gl:     regexp.MustCompile("var"),
-	glName: regexp.MustCompile("(^|[\\s])var\\s([_a-zA-Z][_a-zA-Z0-9]*)"),
+	glName: regexp.MustCompile(`(^|[\s])var\s([_a-zA-Z][_a-zA-Z0-9]*)`),
 
 	bodyOpen:  regexp.MustCompile("{"),
 	bodyClose: regexp.MustCompile("}"),
 
 	imp:     regexp.MustCompile("import"),
-	impName: regexp.MustCompile("(^|[\\s])import\\s+\"([_a-zA-Z][_a-zA-Z0-9/-]*)\""),
+	impName: regexp.MustCompile(`(^|[\s])import\\s+"([_a-zA-Z][_a-zA-Z0-9/-]*)"`),
 }
 
 // lg contains loggers
@@ -343,7 +343,7 @@ func ParseSourceCode(sourceCode []*os.File, fileNames []string) int {
 	for i, source := range sourceCode {
 		tmp := bytes.NewBuffer(nil)
 		io.Copy(tmp, source)
-		sourceCodeCopy[i] = string(tmp.Bytes())
+		sourceCodeCopy[i] = tmp.String()
 	}
 
 	// We need to traverse the elements by hierarchy first add all the

--- a/cxgo/parser/lexer.go
+++ b/cxgo/parser/lexer.go
@@ -4,7 +4,7 @@ import (
 	"fmt"
 	"io"
 
-	. "github.com/skycoin/cx/cxgo/actions"
+	"github.com/skycoin/cx/cxgo/actions"
 )
 
 var DebugLexer bool
@@ -17,7 +17,7 @@ func (yylex Lexer) Error(msg string) {
 func (yylex *Lexer) Lex(lval *yySymType) int {
 	yylex.next()
 	lval.scancopy(yylex.tok)
-	LineNo = lval.line
+	actions.LineNo = lval.line
 	return lval.yys
 }
 
@@ -194,13 +194,13 @@ var tokenNames = map[int]string{
 	CONTINUE:     "CONTINUE",
 	TYPE:         "TYPE",
 
-	// Types 
+	// Types
 	BASICTYPE: "BASICTYPE",
-	// Selectors 
+	// Selectors
 	SPACKAGE: "SPACKAGE",
 	SSTRUCT:  "SSTRUCT",
 	SFUNC:    "SFUNC",
-	// Removers 
+	// Removers
 	REM:     "REM",
 	DEF:     "DEF",
 	EXPR:    "EXPR",
@@ -208,21 +208,21 @@ var tokenNames = map[int]string{
 	CLAUSES: "CLAUSES",
 	OBJECT:  "OBJECT",
 	OBJECTS: "OBJECTS",
-	// Stepping 
+	// Stepping
 	STEP:  "STEP",
 	PSTEP: "PSTEP",
 	TSTEP: "TSTEP",
-	// Debugging 
+	// Debugging
 	DSTACK:   "DSTACK",
 	DPROGRAM: "DPROGRAM",
 	DSTATE:   "DSTATE",
-	// Affordances 
+	// Affordances
 	AFF:   "AFF",
 	CAFF:  "CAFF",
 	TAG:   "TAG",
 	INFER: "INFER",
 	VALUE: "VALUE",
-	// Pointers 
+	// Pointers
 	ADDR: "ADDR",
 }
 */

--- a/cxgo/webapi/httputil.go
+++ b/cxgo/webapi/httputil.go
@@ -2,8 +2,8 @@ package api
 
 import (
 	"encoding/json"
-	"net/http"
 	"fmt"
+	"net/http"
 )
 
 // WriteJSON writes a json object on a http.ResponseWriter with the given code,
@@ -14,7 +14,7 @@ func WriteJSON(w http.ResponseWriter, r *http.Request, code int, v interface{}) 
 	enc := json.NewEncoder(w)
 	pretty, err := BoolFromQuery(r, "pretty", false)
 	if err != nil {
-		fmt.Errorf("Failed to get bool from query")
+		fmt.Fprint(w, "Failed to get bool from query")
 	}
 	if pretty {
 		enc.SetIndent("", "  ")

